py/server/deephaven/calendar.py

```diff
@@ -1,509 +1,136 @@
 #
-# Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+# Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
 #
 
-""" This module provides a collection of business calendars and convenient calendar functions. """
+""" This module defines functions for working with business calendars. """
 
-from enum import Enum
-from typing import List
+from typing import Optional, Union, List
 
 import jpy
 
-from deephaven._wrapper import JObjectWrapper
-from deephaven.time import TimeZone, Instant
-
 from deephaven import DHError
+from deephaven.dtypes import BusinessCalendar
+
 
 _JCalendars = jpy.get_type("io.deephaven.time.calendar.Calendars")
-_JCalendar = jpy.get_type("io.deephaven.time.calendar.Calendar")
-_JDayOfWeek = jpy.get_type("java.time.DayOfWeek")
-_JBusinessPeriod = jpy.get_type("io.deephaven.time.calendar.BusinessPeriod")
-_JBusinessSchedule = jpy.get_type("io.deephaven.time.calendar.BusinessSchedule")
-_JBusinessCalendar = jpy.get_type("io.deephaven.time.calendar.BusinessCalendar")
 
 
-def calendar_names() -> List[str]:
-    """ Returns the names of all available calendars.
+def remove_calendar(name: str) -> None:
+    """ Removes the calendar with the given name from the set of available options.
 
-    Returns:
-        a list of names of all available calendars
+    Args:
+        name (str): the name of the calendar to remove
 
     Raises:
         DHError
     """
     try:
-        return list(_JCalendars.calendarNames())
+        _JCalendars.removeCalendar(name)
     except Exception as e:
-        raise DHError(e, "failed to obtain the available calendar names.") from e
+        raise DHError(e, f"failed to remove calendar '{name}'") from e
 
 
-def default_calendar_name():
-    """ Returns the default calendar name which is set by the 'Calendar.default' property in the configuration file
-    that the Deephaven server is started with.
+def add_calendar(cal: Optional[BusinessCalendar] = None, file: Optional[str] = None) -> None:
```

Comment:
```suggestion
def add_calendar(cal: Union[BusinessCalendar，str]) -> None:
```
I'd prefer this signature design which saves us from having to do all the check on the input below.

Reply:
I'm hesitant to make this change without discussion.  What if we want to add the ability to add a calendar from a json string or something like that?  Then things get funky.

Reply:
Not sure if we need to worry about the specific case of JSON, since`json.load` returns a dict by default, so an additional dict type in the union would work nicely. IMO what I suggested is more Pythonic but it is just a suggestion.